src/Monolog/Handler/WebRequestRecognizerTrait.php

```diff
@@ -0,0 +1,16 @@
+<?php declare(strict_types=1);
+
+namespace Monolog\Handler;
+
+trait WebRequestRecognizerTrait
+{
+
+    /**
+     * Checks if PHP's serving a web request
+     * @return bool
+     */
+    protected function isWebRequest(): bool
+    {
+        return 'cli' !== php_sapi_name();
```

Comment:
use `'cli' !== \PHP_SAPI && 'phpdbg' !== \PHP_SAPI` instead. The `\PHP_SAPI` will be resolved at compile time (when fully qualifying it), which will allow OPCache to resolve the full condition. `php_sapi_name` does not benefit from this AFAICT (especially when not fully qualified).

And the `phpdbg` SAPI is not a web one either.